### ISSUE TITLE
Throws error if used under the strict mode

### ISSUE THREAD

**Issue by lazutkin** [LearnBoost/mongoose#2281]
Try this to reproduce (3.8.x)

```
node --use_strict

> require('mongoose')
crypto.js:538
    function next(er, ret) {
    ^^^^^^^^
SyntaxError: In strict mode code, functions can only be declared at top level or immediately within another function.
    at runInThisContext (node.js:734:18)
    at NativeModule.compile (node.js:797:14)
    at Function.NativeModule.require (node.js:767:18)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:357:17)
    at require (module.js:373:17)
    at Object.<anonymous> (/.../mongoose/node_modules/mongodb/lib/mongodb/commands/db_command.js:5:12)
    at Module._compile (module.js:449:26)
    at Object.Module._extensions..js (module.js:467:10)
    at Module.load (module.js:349:32)
    at Function.Module._load (module.js:305:12)
    at Module.require (module.js:357:17)
```

**Comment by vkarpov15** [comment]
Actually, this is an issue with [NodeJS' native crypto module](https://github.com/joyent/node/blob/master/lib/crypto.js). FWIW I agree that hoisted function definitions are dangerous, so I'll put in a PR and see if they accept it :)

**Comment by vkarpov15** [comment]
Will be fixed in mongoose 4.0 and node 0.12.1. The issue you saw was due to the aforementioned NodeJS 0.12.0 issue, but there were a couple other issues in some of mongoose's dependent modules, including the horribly outdated hooks module. Mongoose 4 should work fine in strict mode now.
